packages/core/core/src/Graph.js

```diff
@@ -154,4 +198,128 @@ export default class Graph {
 
     return {removed, added};
   }
+
+  traverse(
+    visit: (node: Node, context?: any, traversal: TraversalContext) => any,
+    startNode: ?Node
+  ) {
+    return this.dfs({
+      visit,
+      startNode,
+      getChildren: this.getNodesConnectedFrom.bind(this)
+    });
+  }
+
+  traverseAncestors(
+    startNode: Node,
+    visit: (node: Node, context?: any, traversal: TraversalContext) => any
```

Comment:
Most of the other traversal methods have visit as the first arg, is there a reason why this one is different?

Reply:
well... in the other ones it is optional, whereas this one it is required. So I felt like it looked nicer to have the callback at the end in this case.